lib/execution/execution-triple.ts

```diff
@@ -0,0 +1,64 @@
+import os from 'os';
+
+import {BaseExecutionTriple, ExecutionSpecialty} from '../../types/execution/execution-triple.js';
+import {InstructionSet} from '../../types/instructionsets.js';
+
+let _host_specialty = ExecutionSpecialty.cpu;
+
+export function setHostSpecialty(value: ExecutionSpecialty) {
+    _host_specialty = value;
+}
+
+export class ExecutionTriple extends BaseExecutionTriple {
+    constructor() {
+        super();
+        this.initHostTriple();
+    }
+
+    private initHostTriple() {
+        const hostArch = os.arch();
+        if (hostArch === 'x64') {
```

Comment:
Maybe `switch` here? (and below, if it makes sense).

And/or extract to a function to be shared with `matchesCurrentHost()` (or refacto like I suggest to avoid that). Then you don't have to encode `hostArch === `x86` means "instuctionSet amd64in two places; here and in `matchesCurrentHost`